references/classification/utils.py

```diff
@@ -400,3 +401,57 @@ def reduce_across_processes(val):
     dist.barrier()
     dist.all_reduce(t)
     return t
+
+
+def set_weight_decay(
+    model: torch.nn.Module,
+    weight_decay: float,
+    norm_weight_decay: Optional[float] = None,
+    bias_weight_decay: Optional[float] = None,
+    custom_keys_weight_decay: Optional[Dict[str, float]] = None,
+):
+    norm_classes = (torch.nn.modules.batchnorm._BatchNorm, torch.nn.LayerNorm, torch.nn.GroupNorm)
+
+    norm_params = []
+    bias_params = []
+    other_params = []
+    custom_params = {}
+    if custom_keys_weight_decay is not None:
+        for key in custom_keys_weight_decay:
+            custom_params[key] = []
+
+    for module in model.modules():
+        if next(module.children(), None):
+            for name, p in module.named_parameters(recurse=False):
+                if not p.requires_grad:
+                    continue
+                is_custom_key = False
+                for key in custom_params:
+                    if key in name:
```

Comment:
I think we would need to discuss pros/cons of what the API expects the custom_keys to be. I understand you are looking for substring here. Why not require the user to directly define the full names of the param they want to exclude? Looks clearner; thoughts? 

Also it's worth checking if we can simplify the logic here.

Reply:
I agree with this, it could be dangerous to just require a substring, but I might be missing something.

Reply:
you are right, in fact we can use `==` instead of  `in`, because the `name` here means the child name (not include the parent prefix).